docs/rfcs/0002-session-id-sharing.md

```diff
@@ -0,0 +1,223 @@
+and can be joined using that identifier.
+
+On mobile environments, due to the lack of guarantees on the main application process being alive
+throughout most or any of the VPN session, having the main application manage the session_id is unreliable.
+This identifier needs to be rotated on every session start/end and the main application process may
+not be running at these times.
+
+Moreover, on mobile platforms, the fact that the main application process is killed whenever
+the application goes to the background makes the main application telemetry insufficient for collecting
+all the telemetry required to understand the health of the Mozilla VPN application and its usage patterns.
+To address this issue, mobile daemons have a separate instance of Glean, used to collect telemetry
+while the application is backgrounded. In order to get a full picture of a mobile user's session,
+joining the daemon data with the main application is necessary. However, the telemetry clients in
+each process are independent so manually sharing an identifier is required for such joins.
+
+On [VPN session pings - Tech Spec](https://mozilla-hub.atlassian.net/wiki/spaces/SECPRV/pages/184419319/VPN+session+pings+-+Tech+spec)
+an installation_id metric is introduced that would allows joins between daemon and main application
+data. That still does not make it possible to join specific sessions i.e. we are able to infer that
+a given daemon session is coming from the same installation as a main application session, but we
+cannot reliably or easily infer that a given daemon session is the same as a given main application session.
+
+## Document Purpose
+
+Propose a solution to reliably join mobile session data between daemon and main application telemetry.
+
+## Requirements
+
+The main transport for session related data are the session pings, namely
+the [`vpnsession`](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/pings/vpnsession) ping
+and the [`daemonsession`](https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/pings/daemonsession) ping.
+
+It is a requirement on the proposed solution, that the metrics included in each ping are all
+recorded during the session related to the shared session id in it i.e. the ping does not contain
+any metric from a previous session.
+
+## Proposed Solution
+
+A new metric will be introduced to mobile platforms: `shared_session_id`. This metric will be the
+primary session identifier metric for Mozilla VPN mobile telemetry. This metric will have lifetime
+`User`, with this lifetime, once a metric is set it can only be overwritten but never cleared from
+the telemetry storage.
+
+The daemon process will be responsible for managing this metric. The daemon is the preferred
+place for this, because it is _guaranteed_ to be running throughout the whole VPN session.
+
+The main application will communicate with the daemon in order to get the correct value for
```

Comment:
Can we add a couple lines about how the main application will communicate with the daemon and get the UUID back? I think this is already being used for both Android and iOS (and wouldn't require much new code), but would be good to call out in this doc.

Reply:
I am not sure it is necessary to add such technical details to this document. Passing data back from daemon to main app is already implemented in both platforms we will just use the already existing infrastructure. Do you have a specific reason for this ask?

Reply:
These sorts of details can sometimes be things that add lots of complexity to an implementation (if it's not already solved or known how we'll do it, as I now know it is in this situation). While the document still lacks that level of detail, your comment puts my concerns to rest.